src/rgw/rgw_asio_frontend.cc

```diff
@@ -1105,12 +1082,12 @@ void AsioFrontend::join()
   if (!going_down) {
     stop();
   }
-  work.reset();
+
+  // TODO It would be really nice if we could find a way to close the
+  // frontend without killing the entire io_context.
 
   ldout(ctx(), 4) << "frontend joining threads..." << dendl;
-  for (auto& thread : threads) {
-    thread.join();
-  }
+  context_pool.finish();
```

Comment:
it seems like `AsioFrontend::join()` could just be a noop. if the context pool needs to be stopped, `AppMain` should do it

Reply:
Done